src/site/content/en/blog/image-support-for-async-clipboard/index.md

```diff
@@ -0,0 +1,316 @@
+Chrome 76 has added support for images makes it easy to programmatically
+copy and paste **PNG** images.
+
+{% Aside 'caution' %}
+  At the time of writing, only PNG files are supported. Support for other
+  images and file formats will be added in the future.
+{% endAside %}
+
+Before we dive in, let's take a brief look at how the Asynchronous Clipboard
+API works. If you remember the details, skip ahead to the [image section](#images).
+
+## Recap of the Asynchronous Clipboard API {: #recap }
+
+Before describing image support, I want to review how the Asynchronous Clipboard
+API works. Feel free to [skip ahead](#images) if you're already comfortable
+using the API.
+
+### Copy: writing text to the clipboard {: #copy-text }
+
+To copy text to the clipboard, call `navigator.clipboard.writeText()`.
+Since this API is asynchronous, the `writeText()` function returns a promise
+that resolves or rejects depending on whether the passed text is
+copied successfully. For example:
+
+```js
+async function copyPageURL() {
+  try {
+    await navigator.clipboard.writeText(location.href);
+    console.log('Page URL copied to clipboard');
+  } catch (err) {
+    console.error('Failed to copy: ', err);
+  }
+}
+```
+
+### Paste: reading text from the clipboard {: #reading-text }
+
+Much like copy, text can be read from the clipboard by calling
+`navigator.clipboard.readText()` and waiting for the returned promise to
+resolve with the text:
+
+```js
+async function getClipboardText() {
+  try {
+    let text = await navigator.clipboard.readText();
+    console.log('Clipboard contents: ', text);
+  } catch (err) {
+    console.error('Failed to read clipboard contents: ', err);
+  }
+}
+```
+
+### Handling paste events
+
+Paste events can be handled by listening for the (surprise) `paste` event.
+Note that you need to call `preventDefault()` in order to modify the
+to-be-pasted data, like for example, convert it to uppercase before pasting.
+It works nicely with the new asynchronous methods for reading clipboard text:
+
+```js
+document.addEventListener('paste', async (e) => {
+  e.preventDefault();
+  try {
+    const text = await navigator.clipboard.readText();
+    text = text.toUpperCase();
+    console.log('Pasted UPPERCASE text: ', text);
+  } catch (err) {
+    console.error('Failed to read clipboard contents: ', err);
+  }
+});
+```
+
+### Security and permissions {: #security-permission }
+
+The `navigator.clipboard` property is only supported for pages served over
+HTTPS, and to help prevent abuse, clipboard access is only allowed when a page
+is the active tab. Pages in active tabs can write to the clipboard without
+requesting permission, but reading from the clipboard always requires
+permission.
+
+When the Asynchronous Clipboard API was introduced,
+two new permissions for copy and paste were added to the
+ [Permissions API](https://developers.google.com/web/updates/2015/04/permissions-api-for-the-web):
```

Comment:
Should we switch this to an MDN link? https://developer.mozilla.org/en-US/docs/Web/API/Permissions_API

@kaycebasques ?

Reply:
Yes